packages/framework/collabspaces/src/collabSpaces.ts

```diff
@@ -666,12 +667,18 @@ export class TempCollabSpaceRuntime
 		// generate new ID for a columns
 		while (count > 0) {
 			count--;
-			this.matrix.setCell(0, col, uuid() as unknown as MatrixInternalType);
+			const colId = uuid();
+			this.matrix.setCell(0, col, colId as unknown as MatrixInternalType);
+			this.colMap.set(colId, col);
```

Comment:
The problem with doing it here is that it only works for local changes.
We need to ensure we update mappings on both local & remote changes.
For that reason, it would be better if such updates where in response to events that fire in response to changes, and thus capture both local & remote changes